lib/xnvme_be_spdk_dev.c

```diff
@@ -551,9 +551,9 @@ enumerate_attach_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 		    struct spdk_nvme_ctrlr *ctrlr, const struct spdk_nvme_ctrlr_opts *ctrlr_opts)
 {
 	struct xnvme_be_spdk_enumerate_ctx *ectx = cb_ctx;
-	const int num_ns = spdk_nvme_ctrlr_get_num_ns(ctrlr);
+	int nsid = spdk_nvme_ctrlr_get_first_active_ns(ctrlr);
 
-	for (int nsid = 1; nsid <= num_ns; ++nsid) {
+	while (nsid) {
```

Comment:
How about this instead:
``for (int nsid = spdk_nvme_ctrlr_get_first_active_ns(ctrlr); nsid; nsid = spdk_nvme_ctrlr_get_next_active_ns(ctrlr, nsid)) {``

Thus, keeping the ``nsid`` local..

Reply:
Done